dev/flake-module.nix

```diff
@@ -38,11 +44,18 @@
       in tests.runTests pkgs.emptyFile // { internals = tests; };
 
   };
-  flake = {
-    # for repl exploration / debug
-    config.config = config;
-    options.mySystem = lib.mkOption { default = config.allSystems.${builtins.currentSystem}; };
-    config.effects = withSystem "x86_64-linux" ({ pkgs, hci-effects, ... }: {
+  config.flake = { config, options, specialArgs, ... } @ args: {
+    # for REPL exploration / debugging
+    config.allFlakeModuleArgs = args // config._module.args // specialArgs;
+    config.allRootModuleArgs = rootArgs // rootConfig._module.args // rootSpecialArgs;
+    config.transformFlakeModule = flakeModuleTransformer:
+      rootSpecialArgs.replaceSpecialArgs (prevSpecialArgs: prevSpecialArgs // {
+        inherit flakeModuleTransformer;
+      });
+    options.mySystem = lib.mkOption {
+      default = rootConfig.allSystems.${builtins.currentSystem};
+    };
```

Comment:
Can't we just set `config.debug = true;`?

`dev` is already overly complicated because it tries to work around the subflake problems. I'd like to simplify it instead of obfuscating it further.